src/model/resolvable.ts

```diff
@@ -18,9 +18,13 @@ import {Feature, ScannedFeature} from './feature';
 /**
  * A ScannedFeature that needs to resolve other Features to build its final
  * representation.
+ *
+ * TODO(aomarks) Is it right that I made this an abstract class? ScannedFeature
```

Comment:
The only reason to have a pure abstract class rather than an interface is to be able to inherit from it to get it on the prototype chain and use `instanceof`. Since we want to move away from `instanceof` because node/npm, I'm not sure there is a good reason here.

An interface extending a class isn't a weird thing in TypeScript, where every class also defines an implicit interface. This is the same as if we defined both an interface and base class for ScannedFeature and Resolvable extended the interface. I do question whether this interface actually needs to extend ScannedFeature though.